docs/src/pages/api.md

```diff
@@ -241,16 +241,62 @@ Another common way of storing the same data is as a 3×`N` `Matrix{Float64}`.
 Rather conveniently, such types have *exactly* the same binary layout in memory,
 and therefore we can use `reinterpret` to convert between the two formats
 ```julia
-function svectors(x::Matrix{Float64})
-    @assert size(x,1) == 3
-    reinterpret(SVector{3,Float64}, x, (size(x,2),))
+function svectors(x::Matrix{T}, ::Val{N}) where {T,N}
+    size(x,1) == N || throw("sizes mismatch")
+    isbitstype(T) || throw("use for bitstypes only")
+    reinterpret(SVector{N,T}, reshape(x, length(x)))
 end
 ```
-Such a conversion does not copy the data, rather it refers to the *same* memory
-referenced by two different Julia `Array`s. Arguably, a `Vector` of `SVector`s
-is preferable to a `Matrix` because (a) it provides a better abstraction of the
-objects contained in the array and (b) it allows the fast *StaticArrays* methods
-to act on elements.
+Such a conversion does not copy the data, rather it refers to the *same* memory. 
+Arguably, a `Vector` of `SVector`s is often preferable to a `Matrix` because 
+(a) it provides a better abstraction of the objects contained in the array and 
+(b) it allows the fast *StaticArrays* methods to act on elements
+
+However, the resulting object is a Base.ReinterpretArray, not an Array, which may carry some
+runtime penalty on every single access. If you can afford the memory for a copy and can live with
+the non-shared mutation semantics, then it is better to pull a copy by e.g.
+```julia
+function svectorscopy(x::Matrix{T}, ::Val{N}) where {T,N}
+    size(x,1) == N || throw("sizes mismatch")
+    isbitstype(T) || throw("use for bitstypes only")
+    res = Vector{SVector{N,T}}(undef, size(x,2))
+    GC.@preserve res x ccall(:memcpy, Ptr{Cvoid}, (Ptr{Cvoid}, Ptr{Cvoid}, Csize_t), pointer(res), pointer(x), sizeof(x))
```

Comment:
This seems unnecessarily low level for the documentation. How about simply using

`copy(reinterpret(SVector{N,T}, vec(x)))`

If you do think we need this low level optimization perhaps we should be providing some version of `svectors` as part of the official API rather than suggesting users do these low level hacks themselves? (#496 I guess)